src/internal_events/elasticsearch.rs

```diff
@@ -20,11 +19,11 @@ impl InternalEvent for ElasticSearchEventReceived {
 }
 
 #[derive(Debug)]
-pub struct ElasticSearchMissingKeys {
-    pub keys: Vec<Atom>,
+pub struct ElasticSearchMissingKeys<'a> {
+    pub keys: &'a [String],
```

Comment:
Allowing this instead of a `&'a [&'a str]` removes the need for the code that calls this to make an extra allocation through a `by_ref()` call.